images/image.pkr.hcl

```diff
@@ -0,0 +1,36 @@
+variable "region" {
+  type    = string
+  default = "us-east-1"
+}
+
+# source blocks are generated from your builders; a source can be referenced in
+# build blocks. A build block runs provisioners and post-processors on a
+# source.
+source "amazon-ebs" "example" {
+  ami_name      = "learn-terraform-packer"
```

Comment:
This fails with an error. The JSON version used a timestamp in the name in order to generate a (mostly) unique name. Maybe this needs `{{timestamp}}` too?

```
==> amazon-ebs.example: Error: AMI Name: 'learn-terraform-packer' is used by an existing AMI: ami-0dbaca5d269497603
Build 'amazon-ebs.example' errored after 1 second 791 milliseconds: Error: AMI Name: 'learn-terraform-packer' is used by an existing AMI: ami-0dbaca5d269497603
```

Reply:
Updated with timestamp from tutorial text.

Reply:
"timestamp" won't work this way in HCL. You'd need to create a local var: 

`locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }`

and access it like this: 

`"learn-terraform-packer-${local.timestamp}"`

Reply:
Thanks! Just tested and updated.